### DISCUSSION TITLE
Frequency of microcontroller

### DISCUSSION THREAD

**Discussion by amardeepmr40** [olikraus/u8g2#1560] [Q&A]
I'm interfacing my lcd dot matrix of pixel 240*160, but it's too slow on 24mhz frequency. Please guide me on what minimum frequency i can display my image fast. If 24mhz is sufficient then please guide me for refreshing my display for every new image at higher refresh rate.

**Comment by olikraus** [comment]
Much more information is needed here. Which constructor do you use? How much of calculation do you do for the picture?
In general the advice is to do as little calculation as possible in the inner drawing loop.

**Reply by amardeepmr40** [reply]
I want to draw a scroll up or down menu in 160*20 area, please guide how i should do this, i am using avr 8 bit microcontroller and controller st7586s.

**Comment by olikraus** [comment]
There are several examples available, which I suggest to study:

For scrolling:

https://github.com/olikraus/u8g2/blob/master/sys/arduino/u8g2_page_buffer/ScrollingText/ScrollingText.ino

https://github.com/olikraus/u8g2/blob/master/sys/arduino/u8g2_page_buffer/Weather/Weather.ino

For scrolling only on a part of the window, use ClipWindow function:

https://github.com/olikraus/u8g2/blob/master/sys/arduino/u8g2_page_buffer/ClipWindow/ClipWindow.ino

For further speed up, you might also consider to do a partial screen update:

https://github.com/olikraus/u8g2/blob/master/sys/arduino/u8g2_page_buffer/UpdatePartly/UpdatePartly.ino

**Reply by amardeepmr40** [reply]
These all examples are based on c++, kindly suggest me some of c.
Thanks for your reply

**Reply by olikraus** [reply]
U8g2 is mainly an Arduino lib, so the examples are in C++.
On the other side no specific C++ magic is used. You can just replace the C++ functions in the examples by its C counterpart. The u8g2 reference manual lists all the function as C and C++ prototype.

Nevertheless, some C code is the sdl folder, including a scrolling example: https://github.com/olikraus/u8g2/tree/master/sys/sdl/Shennong_2

**Comment by amardeepmr40** [comment]
I am using (u8g2_Setup_st7586s_s028hn118a_f) and AVR , IDE - atmel studio
